src/EditorFeatures/CSharpTest/Diagnostics/FixReturnType/FixReturnTypeTests.cs

```diff
@@ -0,0 +1,183 @@
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics.FixReturnType
+{
+    [Trait(Traits.Feature, Traits.Features.CodeActionsFixReturnType)]
+    public partial class FixReturnTypeTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        internal override (DiagnosticAnalyzer, CodeFixProvider) CreateDiagnosticProviderAndFixer(Workspace workspace)
+            => (null, new CSharpFixReturnTypeCodeFixProvider());
+
+        [Fact]
+        public async Task Simple()
+        {
+            await TestInRegularAndScript1Async(
+@"class C
+{
+    void M()
+    {
+        [|return|] 1;
+    }
+}",
+@"class C
+{
+    int M()
+    {
+        return 1;
+    }
+}");
+        }
+
+        [Fact]
+        public async Task ReturnString()
+        {
+            await TestInRegularAndScript1Async(
+@"class C
+{
+    void M()
+    {
+        [|return|] """";
+    }
+}",
+@"class C
+{
+    string M()
+    {
+        return """";
+    }
+}");
+        }
+
+        [Fact]
+        public async Task ReturnNull()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M()
+    {
+        [|return|] null;
+    }
+}");
+        }
+
+        [Fact]
+        public async Task ReturnC()
+        {
+            await TestInRegularAndScript1Async(
+@"class C
+{
+    void M()
+    {
+        [|return|] new C();
+    }
+}",
+@"class C
+{
+    C M()
+    {
+        return new C();
+    }
+}");
+        }
+
+        [Fact]
+        public async Task ReturnString_AsyncVoid()
+        {
+            await TestInRegularAndScript1Async(
+@"class C
+{
+    async void M()
+    {
+        [|return|] """";
+    }
+}",
+@"class C
+{
+    async System.Threading.Tasks.Task<string> M()
+    {
+        return """";
+    }
+}");
+        }
+
+        [Fact]
+        public async Task ReturnString_AsyncTask()
+        {
+            await TestInRegularAndScript1Async(
+@"class C
+{
+    async System.Threading.Tasks.Task M()
+    {
+        [|return|] """";
+    }
+}",
+@"class C
+{
+    async System.Threading.Tasks.Task<string> M()
+    {
+        return """";
+    }
+}");
+        }
+
+        [Fact]
+        public async Task ReturnString_LocalFunction()
+        {
+            await TestInRegularAndScript1Async(
+@"class C
+{
+    void M()
+    {
+        void local()
+        {
+            [|return|] """";
+        }
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        string local()
+        {
+            return """";
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task ReturnString_AsyncVoid_LocalFunction()
+        {
+            await TestInRegularAndScript1Async(
+@"class C
+{
+    void M()
+    {
+        async void local()
+        {
+            [|return|] """";
+        }
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        async System.Threading.Tasks.Task<string> local()
+        {
+            return """";
+        }
```

Comment:
can we have tests with trivia, validating that updating the return type doesn't mess up trivia?

Reply:
The trivia comes out with an extra newline. That comes from formatting (ie. I think it's an existing issue), so I'll leave it as-is for now.